### ISSUE TITLE
ext4: import error under python 2.7.18

### ISSUE THREAD

**Issue by krisk0** [cubinator/ext4#17]
ext4.py appears to be incompatible with my Python 2
```
# python2 --version
Python 2.7.18
# python2 -c 'import ext4.py'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "ext4.py", line 498
    return f"{type(self).__name__:s}({self.file_block_idx!r:s}, {self.disk_block_idx!r:s}, {self.block_count!r:s})"
                                                                                                                  ^
SyntaxError: invalid syntax
```
I need to use `ext4` under Python2. Please tell me how to fix `ext4.py` so it works.

**Comment by mooseIre** [comment]
你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by cubinator** [comment]
@krisk0 The `SyntaxError` already gives you a hint as to what's going wrong. Namely the [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings "Formatted String Literals on docs.python.org") `f"..."`, which are only available since Python 3.6. You could manually go through the source of `ext4.py` and replace every occurrence of

    f"Some text {expression1:...} more text {expression2:...} more text {expression3:...} ..."

with

    "Some text {var1:...} more text {var2:...} more text {var3:...} ...".format(
        var1 = expression1,
        var2 = expression2,
        var3 = expression3,
        ...
    )

but I am not sure whether I made use of other Python 3 features. I would advise you to take a look at the Python package [`3to2`](https://pypi.org/project/3to2/ "3to2 on PyPI").

**Comment by krisk0** [comment]
Thank you. Will try 3to2
